webapp/sources/rudder/rudder-core/src/main/scala/com/normation/rudder/ncf/TechniqueWriter.scala

```diff
@@ -268,36 +266,58 @@ class TechniqueWriter (
       updatedTechnique <- writeTechnique(technique,methods,modId,committer)
       libUpdate  <- techLibUpdate.update(modId, committer, Some(s"Update Technique library after creating files for ncf Technique ${technique.name}")).
                       toIO.chainError(s"An error occured during technique update after files were created for ncf Technique ${technique.name}")
-      _          <- ZIO.when(doRudderLangTest) {
-                      IOResult.effect(runRudderLangTestLoop(technique)).catchAll(err =>
-                        ApplicationLoggerPure.error("Error when doing rudder-lang test loop. You can disable that test with property " +
-                                                    s"'rudder.lang.test-loop.exec' in rudder config file: ${err.fullMsg}"))
-                    }
     } yield {
       updatedTechnique
     }
   }
 
-  def runRudderLangTestLoop(technique : Technique): Unit = {
-    System.getenv().asScala.get("PATH") match {
-      case Some(path: String) => RunNuCommand.run(Cmd("/opt/rudder/share/rudder-lang/tools/tester.sh", technique.bundleName.value :: technique.category :: Nil, Map("PATH" -> path))).either.runNow match {
-        case Left(error: Error) => ApplicationLogger.error(error.getMessage)
-        case Left(error: RudderError) => ApplicationLogger.error(error.msg)
-        case Right(_)  => ApplicationLogger.info(s"rudder-lang tester successfully looped for technique ${technique.bundleName.value}")
+  def callRudderc(technique : Technique) = {
```

Comment:
I believe that needs to be an independant service, with constructor argument:
- config path file
- rudderc binary path
- target base path

Else we won't be able to test properties on the outputs at all.

Reply:
Regarding testing, we will need to embed a `rudderc` at some point - but I wonder where is the limit between testing `rudderc` and testing integration of `rudderc` into rudder with the specific things for it.

Reply:
(we need at least to test the general output layout, path, etc)